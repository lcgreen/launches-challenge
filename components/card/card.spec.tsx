import React from 'react';
import { render, screen } from '@testing-library/react';
import Card from '.';

const defaultProps = {
  header: <div>Header</div>,
  footer: <div>Footer</div>,
}

describe('Card', () => {
  it('renders the component', async () => {
    const {container} = render(<Card {...defaultProps}>{'children'}</Card>);
    expect(container).toMatchSnapshot()
  });

  it('renders the header', async () => {
    render(<Card {...defaultProps}>{'children'}</Card>);
    expect(screen.getByText('Header')).not.toBeNull();
  });

  it('renders the footer', async () => {
    render(<Card {...defaultProps}>{'children'}</Card>);
    expect(screen.getByText('Footer')).not.toBeNull();
  });

  it('renders the children', async () => {
    render(<Card {...defaultProps}>{'children'}</Card>);
    expect(screen.getByText('children')).not.toBeNull();
  });

});