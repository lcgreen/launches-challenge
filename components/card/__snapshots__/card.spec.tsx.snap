// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders the component 1`] = `
<div>
  <div
    class="container"
    tabindex="0"
  >
    <div
      class="header"
    >
      <div>
        Header
      </div>
    </div>
    <div
      class="body"
    >
      children
    </div>
    <div
      class="footer"
    >
      <div>
        Footer
      </div>
    </div>
  </div>
</div>
`;
